### PR TITLE
fix(dspy): preserve integer precision when loading pandas data

### PR DESCRIPTION
**Pull request by emme1t**
## 📝 Changes Description

`DataLoader.from_pandas()` uses `iterrows()`, which coerces mixed integer/float rows to floating point. Importing an `int64` identifier such as `9007199254740993` alongside a float column produces `9007199254740992`, silently changing the training example.

Convert the frame to object dtype before row iteration so each cell retains its value. Field selection and ordering, input keys, unusual and duplicate column names, and row counts for empty field selections remain covered by regression tests.

Reproduction:

```python
import pandas as pd
from dspy.datasets.dataloader import DataLoader

df = pd.DataFrame({"id": [2**53 + 1], "score": [0.5]})
example = DataLoader().from_pandas(df)[0]
assert int(example.id) == 2**53 + 1
```

Validation on Windows, Python 3.13.13, pandas 3.0.6:

- New regressions on the original implementation: **2 failed, 4 passed**; both failures show rounded integer values.
- DataLoader regressions and Example tests after the fix: **31 passed**, including three duplicate-column cases added during review.
- Adjacent dataset seed tests: **2 passed**.
- Ruff and all applicable configured pre-commit hooks passed.

## ✅ Contributor Checklist

- [ ] Pre-Commit checks are passing (locally and remotely): local hooks passed; remote checks are pending.
- [x] Title of your PR / MR corresponds to the required format.
- [x] Commit message follows required format `{label}(dspy): {message}`.

## ⚠️ Warnings

The new pandas tests use the existing `extra` marker and run with `--extra`.

### CODE REVIEW

File: dspy/datasets/dataloader.py [hunk 1]
```diff
@@ -84,8 +84,10 @@ def from_pandas(
         if fields is None:
             fields = list(df.columns)
 
+        # Keep the index in the tuples so an empty field selection still yields one example per row.
         return [
-            dspy.Example({field: row[field] for field in fields}).with_inputs(*input_keys) for _, row in df.iterrows()
+            dspy.Example(dict(zip(fields, row[1:], strict=True))).with_inputs(*input_keys)
+            for row in df[fields].itertuples(name=None)
```
[the remark this answers is not in the document]
Comment (emme1t) [reply dspy/datasets/dataloader.py]: Addressed in 3680c59. Converting the frame to object dtype before the existing row iteration preserves duplicate-column Series values and exact integer values. Added regressions for default fields, an explicit duplicate label, and a reordered selection; the DataLoader and Example tests now pass all 31 cases.
